packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -5560,17 +5560,69 @@ describe("Teal Opcodes", function () {
       );
     });
 
+    it("uncover: n = 1", () => {
+      push(stack, 4); // stack state = [1, 2, 3, 4]
+      const op = new Uncover(['1'], 1);
+
+      // move top to below 1 element
+      op.execute(stack);
+
+      // stack state = [1, 2, 4, 3]
+      assert.equal(stack.length(), 4);
+
+      assert.equal(stack.pop(), 3n);
+      assert.equal(stack.pop(), 4n);
+      assert.equal(stack.pop(), 2n);
+      assert.equal(stack.pop(), 1n);
+    });
+
+    it("uncover: n = 0 - stack state should not change", () => {
+      push(stack, 4); // stack state = [1, 2, 3, 4]
+
+      const op = new Uncover(['0'], 1);
+      op.execute(stack);
+
+      // stack state = [1, 2, 3, 4]
+      assert.equal(stack.length(), 4);
+
+      assert.equal(stack.pop(), 4n);
+      assert.equal(stack.pop(), 3n);
+      assert.equal(stack.pop(), 2n);
+      assert.equal(stack.pop(), 1n);
```

Comment:
why not using (same in the other test below:
```
assert.deepEqual(stack, [1n, 2n, 3n, 4n])
```

https://www.chaijs.com/api/assert/#method_deepequal

Reply:
We can't because stack and array [1n, 2n, 3n, 4n] is not same type. Shell we create utils function to compare it?

Reply:
I thought we are using array here.

Reply:
Stack is [class](https://github.com/scale-it/algo-builder/blob/master/packages/runtime/src/lib/stack.ts#L9)